festim/material.py

```diff
@@ -7,36 +7,76 @@ class Material:
     Material class
 
     Args:
-        D_0 (float or fem.Constant): the pre-exponential factor of the
+        D_0 (float or dict): the pre-exponential factor of the
             diffusion coefficient (m2/s)
-        E_D (float or fem.Constant): the activation energy of the diffusion
+        E_D (float or dict): the activation energy of the diffusion
             coeficient (eV)
         name (str): the name of the material
 
     Attributes:
-        D_0 (float or fem.Constant): the pre-exponential factor of the
+        D_0 (float or dict): the pre-exponential factor of the
             diffusion coefficient (m2/s)
-        E_D (float or fem.Constant): the activation energy of the diffusion
+        E_D (float or dict): the activation energy of the diffusion
             coeficient (eV)
         name (str): the name of the material
+
+    Usage (1 species):
+        >>> my_mat = Material(D_0=1.9e-7, E_D=0.2, name="my_mat")
+    Usage (multispecies):
+        >>> my_mat = Material(
+                D_0={"Species_1": 1.9e-7, "Species_2": 2.0e-7},
+                E_D={"Species_1": 0.2, "Species_2": 0.3},
+                name="my_mat"
+            )
     """
 
     def __init__(self, D_0, E_D, name=None) -> None:
         self.D_0 = D_0
         self.E_D = E_D
         self.name = name
 
-    def get_diffusion_coefficient(self, mesh, temperature):
+    def get_diffusion_coefficient(self, mesh, temperature, species, model_species):
```

Comment:
You shouldn't have to pass the model species here.

This method should be standalone.

I'll open a PR to your branch to make this change.